### PR TITLE
#167313416 Users should be able to share articles across different channels

### PR DESCRIPTION
**Pull request by nkalyesubula**
#### What does this PR do?

-  It adds functionality for sharing an article over mail and social media channels

#### Description of Task to be completed?

-  As an authenticated user, I should be able to share articles; both mine and others via email and social media platforms.

#### How should this be manually tested?

- clone the repo and then run **npm run dev**
- In your postman, execute this route **http://localhost:PORT/api/v1/articles/{slug}/share/{channel}**.  Change PORT, slug and channel accordingly. Channels can be facebook, twitter or mail.

#### What are the relevant pivotal tracker stories?

- [#167313416](https://www.pivotaltracker.com/story/show/167313416)

#### Screenshots (if appropriate)
Screen Shot 2019-08-28 at 16 47 58
Screen Shot 2019-08-28 at 16 48 55
Screen Shot 2019-08-28 at 16 50 24
Screen Shot 2019-08-28 at 16 50 10

### CODE REVIEW

**Review by Anguandia — DISMISSED** [review]
Great Noah!!, we'll need some sync and you walk us through the process flow.

File: src/controllers/articles.controller.js [hunk 1]
```diff
@@ -209,6 +211,52 @@ class Articles {
...
+   * @memberof Articles
+   */
+  static async shareArticle(req, res) {
+    const article = await db.findOne({
+      where: { slug: req.params.slug }
+    });
+
+    if (!article) {
+      return res.status(404).json({
+        status: 404,
```
Comment (habinezadalvan) [inline src/controllers/articles.controller.js]: Please use util helper

File: src/controllers/articles.controller.js [hunk 2]
```diff
@@ -209,6 +211,52 @@ class Articles {
...
+        message: 'Article is not found.'
+      });
+    }
+    const location = `${process.env.BACKEND_URL}/api/${process.env.API_VERSION}`;
+    const url = `${location}/articles/${req.params.slug}`;
+    switch (req.params.channel) {
+      case 'facebook':
+        await OpenUrlHelper.openUrl(`https:www.facebook.com/sharer/sharer.php?u=${url}`);
+        return res.status(200).json({
```
Comment (habinezadalvan) [inline src/controllers/articles.controller.js]: Please use util helper

File: src/controllers/articles.controller.js [hunk 3]
```diff
@@ -209,6 +211,52 @@ class Articles {
...
+          status: 200,
+          message: `Article shared to ${req.params.channel}`,
+        });
+      case 'twitter':
+        await OpenUrlHelper.openUrl(`https://twitter.com/intent/tweet?url=${url}`);
+        return res.status(200).json({
+          status: 200,
```
Comment (habinezadalvan) [inline src/controllers/articles.controller.js]: Please use util helper

File: src/controllers/articles.controller.js [hunk 4]
```diff
@@ -209,6 +211,52 @@ class Articles {
...
+          message: `Article shared to ${req.params.channel}`,
+        });
+      case 'mail':
+        await OpenUrlHelper.openUrl(`mailto:?subject=${article.title}&body=${url}`);
+        return res.status(200).json({
+          status: 200,
```
Comment (habinezadalvan) [inline src/controllers/articles.controller.js]: Please use util helper

**Review by habinezadalvan — CHANGES_REQUESTED** [review]

**Review by habinezadalvan — CHANGES_REQUESTED** [review]
Please use util helper for every response.

**Review by minega25 — DISMISSED** [review]

**Review by habinezadalvan — DISMISSED** [review]
awesome work.

File: src/controllers/articles.controller.js [hunk 5]
```diff
@@ -195,6 +199,44 @@ class Articles {
...
+      where: { slug: req.params.slug }
+    });
+
+    if (!article) {
+      util.setError(404, 'Article is not found.');
+      return util.send(res);
+    }
+    const location = `${process.env.BACKEND_URL}/api/${process.env.API_VERSION}`;
+    const url = `${location}/articles/${req.params.slug}`;
+    switch (req.params.channel) {
```
Comment (Mnickii) [inline src/controllers/articles.controller.js:223]: Nice switch!

**Review by Mnickii — DISMISSED** [review]
LGTM

**Pull request merged**